packages/inventory/src/Stocktake/DetailView/modal/StocktakeLineEdit/StocktakeLineEdit.tsx

```diff
@@ -17,156 +11,126 @@ import {
   ButtonWithIcon,
   PlusCircleIcon,
   Box,
-  StockLine,
+  Item,
+  ModalMode,
+  DialogButton,
 } from '@openmsupply-client/common';
 import { BatchTable, PricingTable } from './StocktakeLineEditTables';
 import { StocktakeLinePanel } from './StocktakeLinePanel';
-import { createStocktakeRow, wrapStocktakeItem } from './utils';
-import { useStockLines } from '@openmsupply-client/system';
-import { createStocktakeItem } from '../../reducer';
+import { StocktakeLineEditForm } from './StocktakeLineEditForm';
+import { useStocktakeLineEdit } from './hooks';
+import { useSaveStocktakeLines } from 'packages/inventory/src/Stocktake/api';
 
 interface StocktakeLineEditProps {
-  item: StocktakeItem | null;
-  onChangeItem: (item: StocktakeItem | null) => void;
+  item: Item | null;
   mode: ModalMode;
-  draft: StocktakeController;
+  onClose: () => void;
+  isOpen: boolean;
 }
 
 enum Tabs {
   Batch = 'Batch',
   Pricing = 'Pricing',
 }
 
-const createStocktakeLine = (
-  item: StocktakeItem,
-  stockLine: StockLine,
-  countThisLine = true
-): StocktakeLine => {
-  return {
-    stockLineId: stockLine.id,
-    itemCode: item.itemCode(),
-    itemName: item.itemName(),
-    countThisLine,
-    ...stockLine,
-  };
-};
-
 export const StocktakeLineEdit: FC<StocktakeLineEditProps> = ({
   item,
-  draft,
-  onChangeItem,
   mode,
+  onClose,
```

Comment:
TODO: This is re-rendering a little much also - so need to address that!
TODO: The Panel should just use a `DataTable` like inbound - though the tabbing issue is a problem, so not sure if we need a different solution here.
TODO: Handle the "Next" button
TODO: Handle loading states after clicking OK